integration_test.go

```diff
@@ -453,6 +470,198 @@ func setupGRPCServer(t *testing.T, svc *simpleService) (net.Addr, *grpc.Server)
 	return ln.Addr(), s
 }
 
+func TestGRPCStreamBenchmark(t *testing.T) {
+
+	tests := []struct {
+		desc    string
+		opts    Options
+		wantRes string
+		wantErr string
+
+		returnError   error
+		returnOutput  []simple.Foo
+		expectedInput []simple.Foo
+
+		expectStreams        int32
+		expectSentMessage    int32
+		expectReceiveMessage int32
+	}{
+		{
+			desc: "client streaming - without warmup",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/ClientStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{"test":1}{"test":2} {"test":4}`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+				},
+				BOpts: BenchmarkOptions{
+					MaxRequests:    100,
+					WarmupRequests: 0,
+					Connections:    2,
+					Concurrency:    2,
+				},
+			},
+			wantRes:              `Total requests:           100`,
+			returnOutput:         []simple.Foo{{Test: 1}},
+			expectedInput:        []simple.Foo{{Test: 1}, {Test: 2}, {Test: 4}},
+			expectStreams:        100,
+			expectReceiveMessage: 300,
+		},
+		{
+			desc: "client streaming - with warmup",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/ClientStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{"test":1}{"test":2}{"test":-1} {"test":10}`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+				},
+				BOpts: BenchmarkOptions{
+					MaxRequests:    100,
+					WarmupRequests: 1,
+					Connections:    2,
+					Concurrency:    2,
+				},
+			},
+			wantRes:              `Total requests:           100`,
+			returnOutput:         []simple.Foo{{Test: 1}},
+			expectedInput:        []simple.Foo{{Test: 1}, {Test: 2}, {Test: -1}, {Test: 10}},
+			expectStreams:        103,
+			expectReceiveMessage: 412,
```

Comment:
How did you get 412? should it be 404? `100*len(expectedInput_) + 1*len(expectedInput)`. 
Same for `expectStreams`, why 103? (I probably missed something)

Please add a comment for those numbers how they are calculated, it will help to maintain this code

Reply:
benchmark -> `number of benchmark requests` * `len(expectedInput)` (100 * 4) = 400
warmup -> `warmup request` * `len(expectedInput)` (1 * 4) = 4
benchmark connection warmup -> `number of connections in benchmark` *  `len(expectedInput)` (2 * 4) = 8

Total: 412

Sure i'll add a comment